public/components/home/home.controller.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+angular.module('textAnalysis')
+    .controller('homeCtrl', ['$scope', '$http', '$anchorScroll', '$location', 'twitterService',
+        function($scope, $http, $anchorScroll, $location, twitterService) {
+
+            $scope.loadingTweets = false;
+
+            $scope.scrollTo = function(id) {
+                $location.hash(id);
+                $anchorScroll();
+            };
+
+            $scope.userInput = {};
+
+            $scope.loadTweets = function() {
+                $scope.loadingTweets = true;
+                twitterService.loadTweets($scope.userInput.text)
```

Comment:
Before calling loadTweets, I would do a quick check that `$scope.userInput.text` is not an empty string.